### PR TITLE
refactor(sms-erros.js): Solves #5708 removed sms-errors.js

### PR DESCRIPTION
**Pull request by shikhar-scs**
Solves #5708 

As suggested by @shane-tomlinson [here](https://github.com/mozilla/fxa-content-server/issues/5708) all the sub-issues have been fixed which include

1. The entire sms-errors.js file was deleted.

2. The conversion [here](https://github.com/mozilla/fxa-content-server/blob/41add48551e4bd798c6cb9d9cae9a45f9a6f0790/app/scripts/lib/fxa-client.js#L791) was removed in fxa-client.js located at /app/scripts/lib/fxa-client.js

3. The checks/tests [here](https://github.com/mozilla/fxa-content-server/blob/41add48551e4bd798c6cb9d9cae9a45f9a6f0790/app/scripts/views/sms_send.js#L173) were removed from sms_send.js located at /app/scripts/views/sms_send.js

4. The extra tests [here](https://github.com/mozilla/fxa-content-server/blob/41add48551e4bd798c6cb9d9cae9a45f9a6f0790/app/tests/spec/lib/fxa-client.js#L1272) and [here](https://github.com/mozilla/fxa-content-server/blob/master/app/tests/spec/views/sms_send.js#L222:L238) were removed too.

5. As a final measure SmsErrors.toError('THROTTLED') in [this test](https://github.com/mozilla/fxa-content-server/blob/41add48551e4bd798c6cb9d9cae9a45f9a6f0790/app/tests/spec/views/sms_sent.js#L124) was changed to be AuthErrors.toError('THROTTLED').

At 4 places sms-errors had been required and thus, they were removed too (to successfully run grunt lint )and finally authErrors was added at the file mentioned in 5. All of this is reflected in the diff generated.
  

A similar PR was filed by me initially and it has been closed now.

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: e836c2cfae51dbe6190566b9d463c4d3a9dbac94
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by shikhar-scs** [comment]
Also, these alarming no. of commits have been caused because of carelessness and I will squash them once this is ready for merge.

File: .gitignore [hunk 1]
```diff
@@ -17,6 +17,9 @@ results
 npm-debug.log*
 node_modules
 browser_modules
+.idea
+npm-shrinkwrap.json
+package.json
```
Comment (vladikoff) [inline .gitignore]: Please do not modify gitignore

Comment (shikhar-scs) [reply .gitignore]: yeah sure I keep forgetting that

**Comment by vladikoff** [comment]
Cleaned up in https://github.com/mozilla/fxa-content-server/pull/5810 

Thanks @shikhar-scs !!

**Comment by shikhar-scs** [comment]
@vladikoff Thats great, I hope you'll merge all of this in #5810

**Comment by vladikoff** [comment]
@shikhar-scs yeap that's the plan!

**Comment by shikhar-scs** [comment]
Thanx a lot @vladikoff :smile:

**Pull request merged**